rabbit-escape-ui-swing/src/rabbitescape/ui/swing/SwingGameLaunch.java

```diff
@@ -381,6 +387,7 @@ public int addToken( int tileX, int tileY, Token.Type ability )
         {
             graphics.playSound( "place_token" );
         }
+        solutionRecorder.append( new PlaceTokenAction( tileX, tileY ) );
         return now;
```

Comment:
Weird that we append in both GameUI and here. Might be symptomatic of wider bad design or might be fixable?